### PR TITLE
SNOW-1239792: disable oob telemetry in stored procedure

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

skip oob telemetry tests as stored proc does not support oob telemetry

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Review by sfc-gh-sfan — APPROVED** [review]

File: tests/mock/test_oob_telemetry.py [hunk 1]
```diff
@@ -17,6 +17,13 @@
 from snowflake.snowpark.mock._connection import MockServerConnection
 from snowflake.snowpark.mock._telemetry import LocalTestOOBTelemetryService
 from snowflake.snowpark.session import Session
+from tests.utils import IS_IN_STORED_PROC
+
+pytestmark = [
```
Comment (sfc-gh-sfan) [inline tests/mock/test_oob_telemetry.py:22]: Do you need this list? Can you just do 

```
    pytest.mark.skipif(
        IS_IN_STORED_PROC, reason="OOB Telemetry not available in stored procedure"
    )
```

**Review by sfc-gh-jdu — APPROVED** [review]

**Comment by sfc-gh-stan** [comment]
Is there a successful precommit run?

**Comment by sfc-gh-aling** [comment]
@sfc-gh-stan precommit test is running, will hold the PR until there is a successful running

**Comment by sfc-gh-aling** [comment]
precommit running finished, there are two tests failing unrelated to oob telemetry.
one is related to test code, the other is related to udf infra.
will fix the test code in another PR.

**Pull request merged**